src/coprocessor/dag/rpn_expr/function.rs

```diff
@@ -14,24 +14,24 @@
 //! ```
 //!
 //! You can still call the `foo` function as what it looks. The macro doesn't change the function
-//! itself. Instead, it creates a `foo_fn()` function (simply add `_fn` to the original function
-//! name) that generates an `RpnFn` struct.
+//! itself. Instead, it creates a `foo_fn_meta()` function (simply add `_fn_meta` to the original
+//! function name) that generates an `RpnFnMeta` struct.
 //!
 //! If you needs `EvalContext` or the raw `RpnFnCallPayload`, just put it ahead of the function
 //! parameters, and add `ctx` or `payload` argument to the attribute. For example:
 //!
 //! ```ignore
-//! // This generates `with_context_fn() -> RpnFn`
+//! // This generates `with_context_fn_meta() -> RpnFnMeta`
 //! #[rpn_fn(ctx)]
 //! fn with_context(ctx: &mut EvalContext, param: &Option<Decimal>) -> Result<Option<Int>> {
 //!     // Your RPN function logic
 //! }
 //!
-//! // This generates `with_ctx_and_payload_fn() -> RpnFn`
+//! // This generates `with_ctx_and_payload_fn_meta() -> RpnFnMeta`
 //! #[rpn_fn(ctx, payload)]
 //! fn with_ctx_and_payload(
 //!     ctx: &mut EvalContext,
-//!     payload: RpnFnCallPayload<'_>
+//!     payload: &RpnFnCallPayload<'_>
```

Comment:
If payload is passed in, how could this function know which logical row it is in charge of?

Reply:
This example is an RPN function taking 0 parameters. An RPN function should only take care of its parameters and return only one row. It shouldn't touch the raw_args in the payload.
If an RPN function needs the payload, it's going to access `ret_field_type` in most cases.